flink-runtime/src/main/java/org/apache/flink/runtime/operators/GroupReduceCombineDriver.java

```diff
@@ -170,7 +171,14 @@ public void run() throws Exception {
 
 			// write the value again
 			if (!this.sorter.write(value)) {
-				throw new IOException("Cannot write record to fresh sort buffer. Record too large.");
+				if (oversizedRecordCount == Long.MAX_VALUE) {
+					LOG.debug("Number of oversized record has exceeded MAX Long");
+				} else {
+					++oversizedRecordCount;
+					LOG.debug("Cannot write record to fresh sort buffer. Record too large. Oversized record count: {}", oversizedRecordCount);
+				}
+				// simply forward the record
+				this.output.collect((OUT)value);
```

Comment:
I think this is fine, calling `out.collect((OUT) value)`. The generics are a bit confusing here, I think. OUT is for combine functions really the same thing as IN - they are not allowed to change the type. This should really be simplified, otherwise it is confusing, agreed.